### DISCUSSION TITLE
Mirth Launcher: add field for custom java CLI arguments and login credentials

### DISCUSSION THREAD

**Discussion by tom08zehn** [nextgenhealthcare/connect#5263] [Feature/Enhancement Ideas]
The Launcher should have an extra field for command line arguments and save it with the connection parameters.

Example: to launch Mirth Administrator in a custom timezone: you could add `-J-Duser.timezone=America/Los_Angeles` (see mockup screenshot below).

Also, please could you add 2 fields for username and password and store it with the connection parameters so the user can instantly login?

![image](https://user-images.githubusercontent.com/12202052/176927502-118181af-adba-40b3-908b-6feca55c6074.png)

**Comment by jonbartels** [comment]
Regarding JVM args - Do any of the `vmoptions` files in your MCAL installation directory appear to change the behavior?

```
jonathan.bartels@REM-JonathanBartels Contents % find ./ -name "*.vmop*"
.//java/app/mcadministrator/windows-x64/launch.vmoptions
.//java/app/mcadministrator/macos/launch.vmoptions
.//java/app/mcadministrator/macos/launchnoprofile.vmoptions
.//java/app/mcadministrator/unix/launch.vmoptions
.//java/app/mcadministrator/windows/launch.vmoptions
jonathan.bartels@REM-JonathanBartels Contents % find ./ -name "*.vmop*" -exec cat {} \;
-Xdock:name=Mirth Connect Administrator 4.0.1
jonathan.bartels@REM-JonathanBartels Contents % 
```

I'm on a Mac so that "dock" argument would seem to set the name of the MC instances icon when the MC client runs.

Regarding passwords - I do not think this is a good feature idea. MCAL should not be a password manager, there are many good password managers available already. It would be ideal if MC had the right hooks so that password managers could auto-fill the MC logins. I will admit I do not know how such hooks would work or exist though.

MC also has several SSO integrations such as:
* https://github.com/ChristopherSchultz/mirth-plugins has an open source LDAP and MFA plugin
* NextGen sells their LDAP extension 

Instead of having to remember many MC passwords it would be better to integrate it with an existing user directory.

**Comment by jonbartels** [comment]
https://github.com/nextgenhealthcare/connect/discussions/4547 may be related
